components/suggest/src/schema.rs

```diff
@@ -149,15 +157,157 @@ impl ConnectionInitializer for SuggestConnectionInitializer {
+    );
+
+    CREATE TABLE prefix_keywords(
+        keyword_prefix TEXT NOT NULL,
+        keyword_suffix TEXT NOT NULL DEFAULT '',
+        confidence INTEGER NOT NULL DEFAULT 0,
+        rank INTEGER NOT NULL,
+        suggestion_id INTEGER NOT NULL REFERENCES suggestions(id) ON DELETE CASCADE,
+        PRIMARY KEY (keyword_prefix, keyword_suffix, suggestion_id)
+    ) WITHOUT ROWID;
+
+    CREATE UNIQUE INDEX keywords_suggestion_id_rank ON keywords(suggestion_id, rank);
+
+    CREATE TABLE suggestions(
+        id INTEGER PRIMARY KEY,
+        record_id TEXT NOT NULL,
+        provider INTEGER NOT NULL,
+        title TEXT NOT NULL,
+        url TEXT NOT NULL,
+        score REAL NOT NULL
+    );
+
+    CREATE TABLE amp_custom_details(
+        suggestion_id INTEGER PRIMARY KEY,
+        advertiser TEXT NOT NULL,
+        block_id INTEGER NOT NULL,
+        iab_category TEXT NOT NULL,
+        impression_url TEXT NOT NULL,
+        click_url TEXT NOT NULL,
+        icon_id TEXT NOT NULL,
+        FOREIGN KEY(suggestion_id) REFERENCES suggestions(id) ON DELETE CASCADE
+    );
+
+    CREATE TABLE wikipedia_custom_details(
+        suggestion_id INTEGER PRIMARY KEY REFERENCES suggestions(id) ON DELETE CASCADE,
+        icon_id TEXT NOT NULL
+    );
+
+    CREATE TABLE amo_custom_details(
+        suggestion_id INTEGER PRIMARY KEY,
+        description TEXT NOT NULL,
+        guid TEXT NOT NULL,
+        icon_url TEXT NOT NULL,
+        rating TEXT,
+        number_of_ratings INTEGER NOT NULL,
+        FOREIGN KEY(suggestion_id) REFERENCES suggestions(id) ON DELETE CASCADE
+    );
+
+    CREATE INDEX suggestions_record_id ON suggestions(record_id);
+
+    CREATE TABLE icons(
+        id TEXT PRIMARY KEY,
+        data BLOB NOT NULL,
+        mimetype TEXT NOT NULL
+    ) WITHOUT ROWID;
+
+    CREATE TABLE yelp_subjects(
+        keyword TEXT PRIMARY KEY,
+        record_id TEXT NOT NULL
+    ) WITHOUT ROWID;
+
+    CREATE TABLE yelp_modifiers(
+        type INTEGER NOT NULL,
+        keyword TEXT NOT NULL,
+        record_id TEXT NOT NULL,
+        PRIMARY KEY (type, keyword)
+    ) WITHOUT ROWID;
+
+    CREATE TABLE yelp_location_signs(
+        keyword TEXT PRIMARY KEY,
+        need_location INTEGER NOT NULL,
+        record_id TEXT NOT NULL
+    ) WITHOUT ROWID;
+
+    CREATE TABLE yelp_custom_details(
+        icon_id TEXT PRIMARY KEY,
+        score REAL NOT NULL,
+        record_id TEXT NOT NULL
+    ) WITHOUT ROWID;
+
+    CREATE TABLE mdn_custom_details(
+        suggestion_id INTEGER PRIMARY KEY,
+        description TEXT NOT NULL,
+        FOREIGN KEY(suggestion_id) REFERENCES suggestions(id) ON DELETE CASCADE
+    );
+
+    PRAGMA user_version=16;
+"#;
+
+    /// Test running all schema upgrades from V16, which was the first schema with a "real"
+    /// migration.
+    #[test]
+    fn test_all_upgrades() {
+        let db_file = MigratedDatabaseFile::new(SuggestConnectionInitializer, V16_SCHEMA);
+        db_file.run_all_upgrades();
```

Comment:
Do we need any post-upgrade checks here? Will `run_all_upgrades()` just fail automatically? The function signature suggests it returns nothing.

Reply:
I think it should panic if an upgrade fails.  I updated the docs to mention that.